trio/_sync.py

```diff
@@ -268,7 +268,7 @@ def acquire_nowait(self) -> None:
         self.acquire_on_behalf_of_nowait(trio.lowlevel.current_task())
 
     @enable_ki_protection
-    def acquire_on_behalf_of_nowait(self, borrower: Task) -> None:
+    def acquire_on_behalf_of_nowait(self, borrower: object) -> None:
```

Comment:
Thought about adding a generic, but that seems somewhat overkill - and would have to be `T|Task` since class methods do stuff like `self.acquire_on_behalf_of(trio.lowlevel.current_task()`. Could maybe have some other type/class that `ThreadPlaceHolder` and user functions inherit from / have a `Protocol`.